src/MSON.h

```diff
@@ -27,378 +27,385 @@
  * Data types in this document define the MSON AST
  */
 
-namespace mson {
+namespace mson
+{
 
-    /** Markdown */
-    typedef mdp::ByteBuffer Markdown;
+/** Markdown */
+typedef mdp::ByteBuffer Markdown;
 
-    /** Literal */
-    typedef std::string Literal;
+/** Literal */
+typedef std::string Literal;
 
-    /**
-     * Kind of Base Type
-     *
-     * This is an internal thing to keep track of what kind of type,
-     * that particular named type or member is sub-typed from
-     */
-    enum BaseType {
-        UndefinedBaseType = 0,     // Undefined
-        PrimitiveBaseType,         // Primitive Types
-        ImplicitPrimitiveBaseType, // Primitive Types (implicit)
-        ObjectBaseType,            // Object Structure Type
-        ImplicitObjectBaseType,    // Object Structure Type (implicit)
-        ValueBaseType,             // Array & Enum Structure Type
-        ImplicitValueBaseType      // Array & Enum Structure Type (implicit)
-    };
+/**
+ * Kind of Base Type
+ *
+ * This is an internal thing to keep track of what kind of type,
+ * that particular named type or member is sub-typed from
+ */
+enum BaseType {
+    UndefinedBaseType = 0, // Undefined
+    PrimitiveBaseType, // Primitive Types
+    ImplicitPrimitiveBaseType, // Primitive Types (implicit)
+    ObjectBaseType, // Object Structure Type
+    ImplicitObjectBaseType, // Object Structure Type (implicit)
+    ValueBaseType, // Array & Enum Structure Type
+    ImplicitValueBaseType // Array & Enum Structure Type (implicit)
+};
+
+/** Named Types base type table */
+typedef std::map<Literal, BaseType> NamedTypeBaseTable;
+
+/** Named Types inheritance table */
+typedef std::map<Literal, std::pair<Literal, mdp::BytesRangeSet> > NamedTypeInheritanceTable;
+
+/** Named Types dependency table */
+typedef std::map<Literal, std::set<Literal> > NamedTypeDependencyTable;
 
-    /** Named Types base type table */
-    typedef std::map<Literal, BaseType> NamedTypeBaseTable;
+/** A simple or actual value */
+struct Value {
+
+    /** Constructor */
+    Value() : variable(false)
```

Comment:
Additional I would like to allow AllowShortFunctionsOnASingleLine

Transform this:
```
    Value() : variable(false)
    {
    }
```
To this:
```
    Value() : variable(false) {}
```